### PR TITLE
Muon results table logs display improvements

### LINKED ISSUE #25825: Muon results table improvements
If the user does a simultaneous fit (say 2 ws) with 1 shared and at least one unshared parameter. We expect the results table to have one col for the shared parameter but two the unshared parameters (with the unique values displayed). 

If a range of logs values occurs for a simultaneous fit we want to give the mean value (and maybe the standard deviation). 

The PR https://github.com/mantidproject/mantid/pull/21813 has (tests 2 and 3) instructions for checking these.

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Improves the handling of logs in the results table of the new Muon GUI. An error is now shown if you try and choose a log that one of the results does not have. Also the value displayed is the time-averaged value only when the experiment is considered "running". For a simultaneous fit the time-average is computed for each input workspace and then the average of this value is taken.

The table also now has the column types set correctly: logs=X, parameters=Y.

It also fixes a bug with missing filtering single-period runs by status for all ISIS logs. The sample logs GUI had been doing this separately but the values from Python were the unfiltered ones. Note that there are still missing unit tests for `ISISRunLogs`. These will be added in a later issue as these are high-priority fixes.

**To test:**

Try out various fits and check different combinations of logs and results to check the column is displaying sensible results. Check the column types are as expected and they can be plotted.

Check out mixing results with different logs (from different instruments for example) and check you get an error. 

Fixes #25825 

*This does not require release notes* because **the new Muon work will be announced as one item.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py [hunk 1]
```diff
@@ -103,11 +117,7 @@ def log_selection(self, existing_selection):
         format matches that of the ListSelectorPresenter class' model.
         """
         selection = {}
-        fits = self._fit_context.fit_list
-        if not fits:
-            return selection
-
-        logs = log_names(fits[0].input_workspace)
+        logs = self._fit_context.log_names(filter_fn=_log_should_be_displayed)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py]: isn't this a function and therefore needs brackets?

File: scripts/test/Muon/fitting_context_test.py [hunk 2]
```diff
@@ -232,6 +276,159 @@ def test_global_parameters_are_captured(self):
...
+        fit = FitInformation(mock.MagicMock(), 'func1',
+                             [fake1.name(), fake2.name()])
+
+        time_average1 = (10 * 5 + 290 * 20) / 300.
+        time_average2 = (75 * 10 + 195 * 30) / 270.
+        all_average = 0.5 * (time_average1 + time_average2)
+        self.assertAlmostEqual(all_average, fit.log_value('ts_1'), places=6)
+
+
+class FittingContextTest(unittest.TestCase):
```
Comment (AnthonyLim23) [inline scripts/test/Muon/fitting_context_test.py]: should we put this into its own file?

File: scripts/test/Muon/results_tab_widget/results_tab_model_test.py [hunk 3]
```diff
@@ -218,59 +171,48 @@ def test_model_creates_fit_selection_given_no_existing_state(self):
...
+            'run_start': [1, True, True],
+        }
+        expected_selection = deepcopy(existing_selection)
+        expected_selection.update({
+            'run_number': [0, False, True],
+            'run_start': [1, True, True],
+            'magnetic_field': [2, False, True],
+        })
+
+        self.assertDictEqual(expected_selection,
```
Comment (AnthonyLim23) [inline scripts/test/Muon/results_tab_widget/results_tab_model_test.py]: sorry, but being a bit slow. Why does the magnetic_field not matter in the expected_selection?

File: scripts/test/Muon/fitting_context_test.py (hunk 2 above)
Comment (martyngigg) [reply scripts/test/Muon/fitting_context_test.py]: Good point. These tests have grown a lot now.

File: scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py (hunk 1 above)
Comment (martyngigg) [reply scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py]: It's pass the function to `log_names` for that to call it so we don't need the brackets here.

File: scripts/test/Muon/results_tab_widget/results_tab_model_test.py (hunk 3 above)
Comment (martyngigg) [reply scripts/test/Muon/results_tab_widget/results_tab_model_test.py]: Ah yes the model for `ListSelectorPresenter` is a dictionary of `{'name': [position, checked, enabled]}`. This is basically saying that as `magnetic_field` is a newly added log that it is in the list but not checked by default.

Comment (martyngigg) [reply scripts/test/Muon/results_tab_widget/results_tab_model_test.py]: I assumed we wouldn't automatically want all of the logs checked?

Comment (AnthonyLim23) [reply scripts/test/Muon/results_tab_widget/results_tab_model_test.py]: ah ok, that makes sense

**Comment by AnthonyLim23** [comment]
cpp check seems to have fallen over

**Comment by AnthonyLim23** [comment]
cpp check seems to have fallen over

**Comment by martyngigg** [comment]
Thanks. I stopped it before as I had some other commits to push to fix some tests.

**Review by AnthonyLim23 — DISMISSED** [review]

**Comment by AnthonyLim23** [comment]
Has failing unit tests

**Comment by martyngigg** [comment]
Thanks. These should be fixed now.

**Comment by martyngigg** [comment]
My push has dismissed the review though

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by AnthonyLim23** [comment]
has a few failing system tests

**Comment by martyngigg** [comment]
I've just restarted them. They look like failures due to memory issues.

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we get this merged please?

**Pull request merged**